pipelines/src/main/scala/com/fulcrumgenomics/str/pipelines/GenotypeFromGroupedBam.scala

```diff
@@ -0,0 +1,101 @@
+import com.fulcrumgenomics.str.tasks.HipStr.{CreateHaploidChromosomeFile, CreateRegionsBed}
+import com.fulcrumgenomics.str.tasks.{FilterRawReadsFromDuplexMolecule, HipStr, ReadGroupPerDuplexMolecularId, StrGenotypeDuplexMolecules}
+import dagr.core.cmdline.Pipelines
+import dagr.core.tasksystem.Pipeline
+import dagr.tasks.DagrDef.{FilePath, PathPrefix, PathToBam, PathToFasta, PathToIntervals}
+
+@clp(
+  description =
+    """
+      |Pipeline to call STR genotypes from a grouped BAM.
+      |
+      |Assigns reads from the same source molecule to their own read group, filters the reads from the same source
+      |duplex molecule to require a minimum number of reads (overall, AB strand, and BA strand), runs HipSTR to call
+      |genotypes jointly across all source duplex molecules (treats each source molecule as its own sample), and
+      |combines the per-source-duplex-molecule genotypes into a final genotype.
+      |
+      |The grouped BAM should be created by fgbio's `GroupReadsByUmi`
+      |
+      |Requires as input the output path prefix given to DuplexConsensusPipeline.
+      |
+      |An interval list of specifying the same regions as input to HipSTR should be given.  The name field should contain
+      |a comma list of values as follows:
+      |  1. the repeat unit length (ex. `3` for the tri-nucleotide repeat `TCATCATCATCA`).
+      |  2. the number of repeat units (ex. `4` for the tri-nucleotide repeat `TCATCATCATCA`).
+      |  3. the name of the STR (ex. D1S1656)
+      |  4. optionally, the expected (known or truth) number of repeat units for allele #1
+      |  5. optionally, the expected (known or truth) number of repeat units for allele #2
+      |
+      |The option `--min-reads` make take take 1-3 values. For example:
+      |
+      |```
+      |FilterRawReadsFromDuplexMolecule ... --min-reads 10 5 3
+      |```
+      |
+      |In each case if fewer than three values are supplied, the last value is repeated (i.e. `80 40` -> `80 40 40`
+      |and `10` -> `10 10 10`.  The first value applies to all the reads from the same source molecule, the second value
+      |to one single-strand consensus, and the last value to the other single-strand consensus. It is required that if
+      |values two and three differ, the _more stringent value comes earlier_.
+    """,
+  group = classOf[Pipelines]
+)
+class GenotypeFromGroupedBam
+(@arg(flag='i', doc="Input SAM or BAM file (from GroupReadsByUmi).") val input: PathToBam,
+ @arg(flag='r', doc="Path to the reference FASTA.")              val ref: PathToFasta,
+ @arg(flag='l', doc="STR regions to analyze.")                   val intervals: PathToIntervals,
+ @arg(flag='o', doc="Path prefix for output files.")             val output: PathPrefix,
+ @arg(flag='M', minElements=1, maxElements=3, doc="The minimum number of raw reads per source molecule.")
+ val minReads: Seq[Int]
+) extends Pipeline(Some(output.getParent)) {
+
+  override def build(): Unit = {
```

Comment:
Looking back at this pipeline after having gone through some of the tools, I wonder if it wouldn't make more sense to parallelize (scatter gather?) by STR here?  I think that would have a couple of advantages:

1. You'd create multiple BAM, but each would have ~20-fold fewer read groups in them.  I imagine this would make the output VCFs from HipSTR more tractable too since you would have fewer samples, and a lot less no-calls.
2. The pipeline would be a lot more parallelizable, so could complete a lot quicker if >> 1 CPU is available